src/fobnail-attester.c

```diff
@@ -189,9 +215,15 @@ static void coap_metadata_handler(struct coap_resource_t* resource, struct coap_
     int ret;
     UsefulBuf ub_meta;
     UsefulBuf ub;
+    UsefulBufC request;
+    UsefulBufC nonce = NULLUsefulBufC;
     struct meta_data meta;
     printf("Received message: %s\n", coap_get_uri_path(in)->s);
 
+    if (coap_get_data(in, &request.len, (const uint8_t**)&request.ptr)) {
```

Comment:
I'd rather see `get_nonce` taking `in` as a parameter and calling this function internally, so it won't have to be repeated for every URI that uses nonce.

Reply:
Done